contracts/cw721-base/src/contract.rs

```diff
@@ -498,16 +506,38 @@ fn parse_approval<A: Api>(api: A, item: StdResult<KV<Expiration>>) -> StdResult<
     })
 }
 
+fn query_tokens<S: Storage, A: Api, Q: Querier>(
+    deps: &Extern<S, A, Q>,
+    owner: HumanAddr,
+    start_after: Option<String>,
+    limit: Option<u32>,
+) -> StdResult<TokensResponse> {
+    let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
+    // TODO: get pagination working on second indexes
+    let _start = OwnedBound::exclusive(start_after.map(Vec::from));
```

Comment:
I see now why `exclusive()` takes an `Option`.

I think there must be a clearer way for doing this, but cannot see it right now. Maybe building and using an `Option<Bound>`? `Bound` can be a struct (not an enum) with a boolean `exclusive` flag, by example. And the helpers take care of setting it... I'm not sure, just thinking out loud.

Reply:
`range` could accept `Option<Bound>` and we remove `Bound::None`. 

Then we replace this with `let _start = start_after.map(|x| Bound::Exclusive(x.into()));`

Before I had a rather verbose match statement to map None as well as Some, so I made the helpers. I guess it is often better just to type a `map` than call a one-line helper function that increases the API surface to learn.

Reply:
> range could accept Option<Bound> and we remove Bound::None.

I think that's a good idea, yes.